### ISSUE TITLE
short options seems to be broken

### ISSUE THREAD

**Issue by ufobat** [arodland/CLI-Osprey#5]
given following example:

```
package Hello {
	use Moo;
	use CLI::Osprey;

	option message => (is => 'ro', short => 'm', format => 's');

	sub run {
		my $self = shift;
		print "Hello " . $self->message . "\n";
	}
}

use strict;
use warnings;

Hello->new_with_options->run();
```

this happens:
```
> perl osprey.pl --message foo
Hello foo
> perl osprey.pl -m foo       
Hello essage
```

With my real world example the errors looks even stranger:
(I've enabled stacktraces, hoping that would help you to troubleshoot this)
```
 perl -Ilib bin/couchconfd import -r .         
Unknown option: e
 at /home/martin/perl5/perlbrew/perls/perl-5.26.0/lib/5.26.0/Getopt/Long.pm line 1072.
	Getopt::Long::FindOption(ARRAY(0x7c4ef0), "(--|-|\\+)", "--", "-ecursive", HASH(0xe85d60)) called at /home/martin/perl5/perlbrew/perls/perl-5.26.0/lib/5.26.0/Getopt/Long.pm line 528
	Getopt::Long::GetOptionsFromArray(undef, undef, undef, undef, undef, undef, undef, undef, ...) called at /home/martin/perl5/perlbrew/perls/perl-5.26.0/lib/site_perl/5.26.0/Getopt/Long/Descriptive.pm line 438
	Getopt::Long::Descriptive::__ANON__("Usage: couchconfd import %o", ARRAY(0x178d658), ARRAY(0x1796f60), ARRAY(0x17972c0), ARRAY(0x1797458), ARRAY(0x1797788), ARRAY(0x1797aa0), ARRAY(0x1795558), ...) called at /home/martin/perl5/perlbrew/perls/perl-5.26.0/lib/site_perl/5.26.0/CLI/Osprey/Role.pm line 228
	CLI::Osprey::Role::parse_options("App::couchconfd::Import", "database", "http://ip:4000/", "command", "kill 1", "permission", 644, "parent_command", ...) called at /home/martin/perl5/perlbrew/perls/perl-5.26.0/lib/site_perl/5.26.0/CLI/Osprey/Role.pm line 143
	CLI::Osprey::Role::new_with_options("App::couchconfd::Import", "destination", "/opt/e2l/etc", "invoked_as", "couchconfd", "user", "e2lservice", "database", ...) called at /home/martin/perl5/perlbrew/perls/perl-5.26.0/lib/site_perl/5.26.0/CLI/Osprey/Role.pm line 196
	CLI::Osprey::Role::new_with_options("App::couchconfd", "permission", 644, "database", "http://ip:4000/", "command", "kill 1", "user", ...) called at lib/App/couchconfd.pm line 30
	App::couchconfd::main("App::couchconfd") called at bin/couchconfd line 7
Unknown option: c
 at /home/martin/perl5/perlbrew/perls/perl-5.26.0/lib/5.26.0/Getopt/Long.pm line 1072.
	Getopt::Long::FindOption(ARRAY(0x7c4ef0), "(--|-|\\+)", "--", "-cursive", HASH(0xe85d60)) called at /home/martin/perl5/perlbrew/perls/perl-5.26.0/lib/5.26.0/Getopt/Long.pm line 528
	Getopt::Long::GetOptionsFromArray(undef, undef, undef, undef, undef, undef, undef, undef, ...) called at /home/martin/perl5/perlbrew/perls/perl-5.26.0/lib/site_perl/5.26.0/Getopt/Long/Descriptive.pm line 438
	Getopt::Long::Descriptive::__ANON__("Usage: couchconfd import %o", ARRAY(0x178d658), ARRAY(0x1796f60), ARRAY(0x17972c0), ARRAY(0x1797458), ARRAY(0x1797788), ARRAY(0x1797aa0), ARRAY(0x1795558), ...) called at /home/martin/perl5/perlbrew/perls/perl-5.26.0/lib/site_perl/5.26.0/CLI/Osprey/Role.pm line 228
	CLI::Osprey::Role::parse_options("App::couchconfd::Import", "database", "http://ip:4000/", "command", "kill 1", "permission", 644, "parent_command", ...) called at /home/martin/perl5/perlbrew/perls/perl-5.26.0/lib/site_perl/5.26.0/CLI/Osprey/Role.pm line 143
	CLI::Osprey::Role::new_with_options("App::couchconfd::Import", "destination", "/opt/e2l/etc", "invoked_as", "couchconfd", "user", "e2lservice", "database", ...) called at /home/martin/perl5/perlbrew/perls/perl-5.26.0/lib/site_perl/5.26.0/CLI/Osprey/Role.pm line 196
	CLI::Osprey::Role::new_with_options("App::couchconfd", "permission", 644, "database", "http://ip:4000/", "command", "kill 1", "user", ...) called at lib/App/couchconfd.pm line 30
	App::couchconfd::main("App::couchconfd") called at bin/couchconfd line 7
Can't locate object method "die" via package "CLI::Osprey::Descriptive::Usage" at /home/martin/perl5/perlbrew/perls/perl-5.26.0/lib/site_perl/5.26.0/Getopt/Long/Descriptive.pm line 438.
	Getopt::Long::Descriptive::__ANON__("Usage: couchconfd import %o", ARRAY(0x178d658), ARRAY(0x1796f60), ARRAY(0x17972c0), ARRAY(0x1797458), ARRAY(0x1797788), ARRAY(0x1797aa0), ARRAY(0x1795558), ...) called at /home/martin/perl5/perlbrew/perls/perl-5.26.0/lib/site_perl/5.26.0/CLI/Osprey/Role.pm line 228
	CLI::Osprey::Role::parse_options("App::couchconfd::Import", "database", "http://ip:4000/", "command", "kill 1", "permission", 644, "parent_command", ...) called at /home/martin/perl5/perlbrew/perls/perl-5.26.0/lib/site_perl/5.26.0/CLI/Osprey/Role.pm line 143
	CLI::Osprey::Role::new_with_options("App::couchconfd::Import", "destination", "/opt/e2l/etc", "invoked_as", "couchconfd", "user", "e2lservice", "database", ...) called at /home/martin/perl5/perlbrew/perls/perl-5.26.0/lib/site_perl/5.26.0/CLI/Osprey/Role.pm line 196
	CLI::Osprey::Role::new_with_options("App::couchconfd", "permission", 644, "database", "http://ip:4000/", "command", "kill 1", "user", ...) called at lib/App/couchconfd.pm line 30
	App::couchconfd::main("App::couchconfd") called at bin/couchconfd line 7

```

**Comment by arodland** [comment]
I see why. I'm working on it. :)

**Comment by arodland** [comment]
Whoops, this seems to have gotten left by the wayside, but I believe the fix is the same as the one for #24 and #26, so this should be resolved as of 0.06.
